src/vs/editor/contrib/referenceSearch/browser/referenceSearch.ts

```diff
@@ -7,7 +7,7 @@
 import * as nls from 'vs/nls';
 import { KeyCode, KeyMod } from 'vs/base/common/keyCodes';
 import URI from 'vs/base/common/uri';
-import { TPromise } from 'vs/base/common/winjs.base';
+import { PPromise, TPromise } from 'vs/base/common/winjs.base';
```

Comment:
Imo we shouldn't use it. Promises with extra sauce are considered an anti pattern by now. 
The function signature is backwards compatible if you make the progress callback optional.
What do you mean with an object that the toggle widget can subscribe to?